client/src/pages/SignUp.jsx

```diff
@@ -34,265 +34,75 @@ const styles = {
 };
 
 export default function SignUp() {
-  // Experimenting with useFormHook
   const [createTeacher, { error }] = useMutation(ADD_TEACHER);
 
-  // destructure
+  const { signInSchema } = useSchemas();
+
   const {
     register,
     handleSubmit,
-    watch,
     formState: { errors },
-  } = useForm();
+  } = useForm({ resolver: yupResolver(signInSchema) });
 
-  // This is where the data will be
   const onSubmit = async (userInput) => {
     console.log(userInput);
     try {
+      // data object returned by Apollo
       const { data } = await createTeacher({
         variables: { ...userInput },
       });
-      Auth.login(data);
+      Auth.login(data); //signing in the user once created
       alert("Account created!");
     } catch (err) {
```

Comment:
Obviously you're not tooled to report exceptions to some aggregator like Sentry, but there are two things I'll say about this bit for posterity:
1.This is a classic "swallowed" error. `catch (err)` here will gobble up ANY exception that occurs inside the try. That could be a syntax error, that could be a network connection error, that could be anything.
2. To avoid swallowing this error here and losing the helpful information it might contain forever, you might be tempted to `console.log(err)` or `SomeErrorReportingTool.report(err)`. But that's potentially problematic as well. Does err contain sensitive data? Could it ever? Do we _know_ what data would be considered sensitive here?

This is all to say-- `try-catch` can be a little bit nasty in Javascript in my view.

Reply:
I definitely see that it is going to gobble up any error but I'm not quite sure what to do as an alternative. Any suggestions? I know in python you can target specific exceptions but I haven't seen anything like that in JavaScript up to this point.